src/Storages/ObjectStorageQueue/ObjectStorageQueueIFileMetadata.cpp

```diff
@@ -260,13 +264,51 @@ void ObjectStorageQueueIFileMetadata::resetProcessing()
     if (state != FileStatus::State::Processing)
         throw Exception(ErrorCodes::LOGICAL_ERROR, "Cannot reset non-processing state: {}", state);
 
-    resetProcessingImpl();
-    file_status->reset();
-}
+    SCOPE_EXIT({
+        file_status->reset();
+    });
 
-void ObjectStorageQueueIFileMetadata::resetProcessingImpl()
-{
-    throw Exception(ErrorCodes::LOGICAL_ERROR, "resetProcessingImpl is not implemented");
+    if (!processing_id_version.has_value())
+    {
+        throw Exception(
+            ErrorCodes::LOGICAL_ERROR,
+            "No processing id version set, but state is `Processing` ({})",
+            node_metadata.toString());
+    }
+
+    Coordination::Requests requests;
+    requests.push_back(zkutil::makeRemoveRequest(processing_node_id_path, processing_id_version.value()));
+    requests.push_back(zkutil::makeRemoveRequest(processing_node_path, -1));
+
+    Coordination::Responses responses;
+    const auto zk_client = getZooKeeper();
+    const auto code = zk_client->tryMulti(requests, responses);
+    if (code == Coordination::Error::ZOK)
+        return;
+
+    if (Coordination::isHardwareError(code))
+    {
+        LOG_TRACE(log, "Keeper session expired, processing will be automatically reset");
+        return;
+    }
+
+    if (responses[0]->error == Coordination::Error::ZBADVERSION)
+    {
+        LOG_WARNING(
+            log, "Processing node no longer exists ({}) "
+            "while setting file as non-retriable failed. "
+            "This could be as a result of expired keeper session. "
+            "Cannot set file as failed, will retry.",
+            processing_node_path);
+        chassert(false);
+        return;
+    }
+
+    auto exception = zkutil::KeeperMultiException(code, requests, responses);
```

Comment:
maybe prefer using `getFailedOpIndex` and directly call `requests[failed_op_idx]->getPath()`
creating an entire `exception` object just to extract a path seems overkill and it confused me